apps/tenant-management-webapp-e2e/src/integration/calendar/calendar.page.ts

```diff
@@ -0,0 +1,89 @@
+class CalendarPage {
+
+  addCalendarOverviewTabBtn() {
+    return cy.get('[data-testid="overall-calendar-add-btn"]');
+  }
+  addCalendarBtn() {
+    return cy.get('[data-testid="add-calendar-btn"]');
+  }
+
+  addScriptModalTitle()
+  {
+    return cy.xpath('//goa-modal[@data-testid="add-calendar-modal" and @open="true"]//*[@slot="heading"]');
+  }
+
+  calendarModalCancelButton()
+  {
+    return cy.get('[data-testid="calendar-modal-cancel"]');
+  }
+
+  calendarModalSaveButton()
+  {
+    return cy.get('[data-testid="calendar-modal-save"]');
+  }
+
+  calendarTableHeader()
+  {
+    return cy.get('[data-testid="calendar-table-header"]');
+  }
+
+  addCalendarModalNameField() {
+    return cy.xpath('//goa-modal[@data-testid="add-calendar-modal" and @open="true"]//*[@data-testid="calendar-modal-name-input"]');
+  }
+
+  addCalendarModalNameFormItem() {
+    return cy.xpath('//goa-modal[@data-testid="add-calendar-modal" and @open="true"]//*[@label="Name"]');
+  }
+
+  addCalendarModalDescriptionField() {
+    return cy.xpath(
+      '//goa-modal[@data-testid="add-calendar-modal" and @open="true"]//*[@data-testid="calendar-modal-description-input"]'
+    );
+  }
+
+  addCalendarModalClientRolesTable(clientName) {
+    return cy.xpath(
+      `//*[@data-testid="add-calendar-modal" and @open="true"]//h4/div[text()="${clientName}"]/parent::h4/following-sibling::goa-table`
+    );
+  }
+
+  addCalendarModalRolesTable() {
+    return cy.xpath(
+      `//*[@data-testid="add-calendar-modal" and @open="true"]//h4[text()="autotest"]/following-sibling::goa-table`
+    );
+  }
+
+  addScriptModalClientRolesTable(clientName) {
+    return cy.xpath(
+      `//*[@data-testid="add-script-modal" and @open="true"]//h4/div[text()="${clientName}"]/parent::h4/following-sibling::goa-table`
+    );
+  }
+
+  calendarTableBody() {
+    return cy.xpath('//table[@data-testid="calendar-table"]//tbody');
+  }
+
+  calendarEditButton(rowNumber) {
+    return cy.xpath(`(//table[@data-testid="calendar-table"]//*[contains(@data-testid, "calendar-edit")])[${rowNumber}]`);
+  }
+
+  calendarDeleteButton(rowNumber) {
+    return cy.xpath(`(//table[@data-testid="calendar-table"]//*[contains(@data-testid, "delete-icon")])[${rowNumber}]`);
+  }
+
+  editCalendarModal() {
```

Comment:
Looks like your edit modal page objects have add-calendar-modal ids. Can they be shared with the corresponding add modal page objects?

Reply:
Yes one can be used for both.